cluster-autoscaler/debuggingsnapshot/debugging_snapshot_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debuggingsnapshot
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+func TestBasicSetterWorkflow(t *testing.T) {
+	snapshot := &DebuggingSnapshotImpl{}
+	pod := []*framework.PodInfo{
+		{
+			Pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "Pod1",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+			},
+		},
+	}
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testNode",
+		},
+	}
+
+	nodeInfo := &framework.NodeInfo{
+		Pods:             pod,
+		Requested:        &framework.Resource{},
+		NonZeroRequested: &framework.Resource{},
+		Allocatable:      &framework.Resource{},
+		Generation:       0,
+	}
+
+	var nodeGroups []*framework.NodeInfo
+	nodeGroups = append(nodeGroups, nodeInfo)
+	nodeGroups[0].SetNode(node)
+	timestamp := time.Now().In(time.UTC)
+	snapshot.SetNodeGroupInfo(nodeGroups)
+	snapshot.SetTimestamp(timestamp)
+	op, err := snapshot.GetAll()
+	assert.False(t, err)
+
+	gkeSnapshot := &DebuggingSnapshotImpl{
+		NodeInfo: []*NodeInfo{
+			{
+				Node: node,
+				Pods: pod,
+			},
+		},
+		Timestamp: timestamp,
+	}
+	expected, err1 := json.Marshal(gkeSnapshot)
+	assert.NoError(t, err1)
+	assert.EqualValues(t, expected, *op)
```

Comment:
This test is essentially comparing json.Marshal output with itself. Maybe instead it would be more useful to Unmarshal op into a `map[string]interface{}` and check if it contains the fields you expect it to contain?

Reply:
done